src/current/_includes/releases/v23.2/v23.2.0-alpha.4.md

```diff
@@ -0,0 +1,197 @@
+- Added two changes to `FOR UPDATE`:
+    - Multiple `FOR UPDATE` clauses on fully parenthesized queries are now disallowed. For example, the following statements are now disallowed:
+
+        {% include_cached copy-clipboard.html %}
+        ~~~ sql
+        (SELECT 1 FOR UPDATE) FOR UPDATE;
+        SELECT * FROM ((SELECT 1 FOR UPDATE) FOR UPDATE) AS x;
+        ~~~
+
+        Whereas statements like the following are still allowed:
+
+        {% include_cached copy-clipboard.html %}
+        ~~~ sql
+        SELECT * FROM (SELECT 1 FOR UPDATE) AS x FOR UPDATE;
+        SELECT (SELECT 1 FOR UPDATE) FOR UPDATE;
+        ~~~
+
+        This does not match PostgreSQL, which allows all of these, but does match CockroachDB behavior for `ORDER BY` and `LIMIT`.
+    - `FOR UPDATE` is now allowed on statements with `VALUES` in the `FROM` list, or as a subquery. For example, the following statements are now allowed:
+
+        {% include_cached copy-clipboard.html %}
+        ~~~ sql
+        SELECT (VALUES (1)) FOR UPDATE;
+        SELECT * FROM (VALUES (1)) AS x FOR UPDATE;
+        ~~~
+
+        Using `FOR UPDATE` directly on `VALUES` is still disallowed:
+
+        {% include_cached copy-clipboard.html %}
+        ~~~ sql
+        VALUES (1) FOR UPDATE; (VALUES (1)) FOR UPDATE;
+        INSERT INTO t VALUES (1) FOR UPDATE;
+        ~~~
+
+        This matches PostgreSQL. [#111258][#111258]
+- `FOR UPDATE` is now permitted on some queries that were previously disallowed. Queries that use the following operations are now allowed to have `FOR UPDATE OF` as long as the prohibited operation is in a subquery not locked by the `FOR UPDATE OF`:
+    - `UNION`
+    - `INTERSECT`
+    - `EXCEPT`
+    - `DISTINCT`
+    - `GROUP BY`
+    - `HAVING`
+    - Aggregations
+    - [Window functions](../v23.2/window-functions.html)
+
+    For example, the following query is now allowed because the subquery using the prohibited operations is not affected by the `FOR UPDATE OF`:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    SELECT * FROM t, (SELECT DISTINCT 0, 0 UNION SELECT a, count(*) FROM t GROUP BY a HAVING a > 0) AS u FOR UPDATE OF t;
+    ~~~
+
+    This matches PostgreSQL. [#111258][#111258]
+- Identifiers after numeric constants that are not separated by whitespace are now disallowed to match PostgreSQL 15 behavior. [#112021][#112021]
+- Added the new column `contention_type` to the [`crdb_internal.transaction_contention_events`](../v23.2/crdb-internal.html) table. This column indicates the type of transaction contention encountered. Current values are `LOCK_WAIT` and `SERIALIZATION_CONFLICT`. These contention events specify that contention occurred between two transactions due to serialization conflicts. [#111685][#111685]
```

Comment:
link "transaction contention" to https://www.cockroachlabs.com/docs/v23.1/performance-best-practices-overview#transaction-contention

you may remove the sentence "These contention events specify that contention occurred between two transactions due to serialization conflicts."

Reply:
done